![Hand with calculator](calculator.jpg)

Did you know that the average return from investing in stocks is [10% per year](https://www.nerdwallet.com/article/investing/average-stock-market-return) (not accounting for inflation)? But who wants to be average?! 

You have been asked to support an investment firm by analyzing trends in high-growth companies. They are interested in understanding which industries are producing the highest valuations and the rate at which new high-value companies are emerging. Providing them with this information gives them a competitive insight as to industry trends and how they should structure their portfolio looking forward.

You have been given access to their `unicorns` database, which contains the following tables:

## dates
| Column       | Description                                  |
|------------- |--------------------------------------------- |
| `company_id`   | A unique ID for the company.                 |
| `date_joined` | The date that the company became a unicorn.  |
| `year_founded` | The year that the company was founded.       |

## funding
| Column           | Description                                  |
|----------------- |--------------------------------------------- |
| `company_id`       | A unique ID for the company.                 |
| `valuation`        | Company value in US dollars.                 |
| `funding`          | The amount of funding raised in US dollars.  |
| `select_investors` | A list of key investors in the company.      |

## industries
| Column       | Description                                  |
|------------- |--------------------------------------------- |
| `company_id`   | A unique ID for the company.                 |
| `industry`     | The industry that the company operates in.   |

## companies
| Column       | Description                                       |
|------------- |-------------------------------------------------- |
| `company_id`   | A unique ID for the company.                      |
| `company`      | The name of the company.                          |
| `city`         | The city where the company is headquartered.      |
| `country`      | The country where the company is headquartered.   |
| `continent`    | The continent where the company is headquartered. |


**The Task**

The task is to first identify the three best-performing industries based on the number of new unicorns created in 2019, 2020, and 2021 combined.

From those industries (1), find the number of unicorns within these industries (2), the year that they became a unicorn (3), and their average valuation, converted to billions of dollars and rounded to two decimal places (4).

**Steps:**


**1.Finding the top industries**

Use a Common Table Expression (CTE) to identify the top performing industries in 2019, 2020, and 2021 combined based on volume of unicorns.



**2.Gathering yearly rankings data**

Build a second CTE to contain the industry, the year, a count of companies aliased as num_unicorns, and the average valuation.


**3.Returning the final results**

Build your query to return the top performing industries in the three target years.


In [41]:

WITH top_list AS(
	SELECT
		industry,
		COUNT(company_id) 
		FROM dates
		INNER JOIN industries
		USING (company_id)
		WHERE EXTRACT(YEAR FROM date_joined) in (2019,2020,2021)
		GROUP BY industry
		ORDER BY count DESC
	    LIMIT 3
	),
avg_valuation AS (
SELECT
EXTRACT(YEAR FROM date_joined) AS year,
industry,
COUNT(company_id) AS num_unicorns,
ROUND(AVG(valuation)/1e9,2) AS average_valuation_billions
FROM dates
INNER JOIN industries USING (company_id)
INNER JOIN funding USING (company_id)
WHERE EXTRACT(YEAR FROM date_joined) in (2019,2020,2021)
GROUP BY year, industry
	)
	
SELECT 
	industry,
	year,
	num_unicorns,
	average_valuation_billions
FROM avg_valuation
INNER JOIN top_list 
USING(industry)
ORDER by year DESC, num_unicorns DESC

industry  ...  average_valuation_billions
0                          Fintech  ...                        2.75
1     Internet software & services  ...                        2.15
2  E-commerce & direct-to-consumer  ...                        2.47
3     Internet software & services  ...                        4.35
4  E-commerce & direct-to-consumer  ...                        4.00
5                          Fintech  ...                        4.33
6                          Fintech  ...                        6.80
7     Internet software & services  ...                        4.23
8  E-commerce & direct-to-consumer  ...                        2.58

[9 rows x 4 columns]

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>industry</th>
      <th>year</th>
      <th>num_unicorns</th>
      <th>average_valuation_billions</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Fintech</td>
      <td>2021</td>
      <td>138</td>
      <td>2.75</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Internet software &amp; services</td>
      <td>2021</td>
      <td>119</td>
      <td>2.15</td>
    </tr>
    <tr>
      <th>2</th>
      <td>E-commerce &amp; direct-to-consumer</td>
      <td>2021</td>
      <td>47</td>
      <td>2.47</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Internet software &amp; services</td>
      <td>2020</td>
      <td>20</td>
      <td>4.35</td>
    </tr>
    <tr>
      <th>4</th>
      <td>E-commerce &amp; direct-to-consumer</td>
      <td>2020</td>
      <td>16</td>
      <td>4.00</td>
    </tr>
    <tr>
      <th>5</th>
      <td>Fintech</td>
      <td>2020</td>
      <td>15</td>
      <td>4.33</td>
    </tr>
    <tr>
      <th>6</th>
      <td>Fintech</td>
      <td>2019</td>
      <td>20</td>
      <td>6.80</td>
    </tr>
    <tr>
      <th>7</th>
      <td>Internet software &amp; services</td>
      <td>2019</td>
      <td>13</td>
      <td>4.23</td>
    </tr>
    <tr>
      <th>8</th>
      <td>E-commerce &amp; direct-to-consumer</td>
      <td>2019</td>
      <td>12</td>
      <td>2.58</td>
    </tr>
  </tbody>
</table>
</div>